eitprocessing/datahandling/mixins/slicing.py

```diff
@@ -54,12 +55,15 @@ def select_by_index(
             warnings.warn("No starting or end timepoint was selected.")
             return self
 
-        start = start or 0
-        end = end or len(self)
-        newlabel = newlabel or f"Slice ({start}-{end}] of <{self.label}>"
+        start = start if isinstance(start, int) else 0
+        end = end if isinstance(end, int) else len(self)
```

Comment:
In the previous iteration, if `end = 0`, it was set to `len(self)` because `bool(0) is False`. (This was one I had to dig deep for while fixing the test.) `end` is 0 in some cases, which should result in an empty copy.

For `start` it does not really matter, but I added it for consistency.

Reply:
I understand and good catch! I can imagine that that took a while to find 😖.

Is this marginally better? It protects against a user inputting something random that is not supposed to be passed here. (Note: for start it also protects against something falsy like an empty string, which `start or 0` would not).

```suggestion
        start = start if start is not None else 0
        end = end if end is not None else len(self)
```